// hume/dist/serialization/resources/customModels/types/FacePrediction.js
"use strict";
/**
 * This file was auto-generated by Fern from our API Definition.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.FacePrediction = void 0;
const core = __importStar(require("../../../../core"));
const BoundingBox_1 = require("./BoundingBox");
const EmotionScore_1 = require("./EmotionScore");
const FacsScore_1 = require("./FacsScore");
const DescriptionsScore_1 = require("./DescriptionsScore");
exports.FacePrediction = core.serialization.object({
    frame: core.serialization.number(),
    time: core.serialization.number(),
    prob: core.serialization.number(),
    box: BoundingBox_1.BoundingBox,
    emotions: core.serialization.list(EmotionScore_1.EmotionScore),
    facs: core.serialization.list(FacsScore_1.FacsScore).optional(),
    descriptions: core.serialization.list(DescriptionsScore_1.DescriptionsScore).optional(),
});
