// hume/serialization/resources/customModels/types/DatasetVersion.js
"use strict";
/**
 * This file was auto-generated by Fern from our API Definition.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.DatasetVersion = void 0;
const core = __importStar(require("../../../../core"));
const DatasetVersionFeatureTypesValue_1 = require("./DatasetVersionFeatureTypesValue");
exports.DatasetVersion = core.serialization.object({
    id: core.serialization.string(),
    userId: core.serialization.property("user_id", core.serialization.string()),
    labelsFileUri: core.serialization.property("labels_file_uri", core.serialization.string()),
    featureTypes: core.serialization.property("feature_types", core.serialization.record(core.serialization.string(), DatasetVersionFeatureTypesValue_1.DatasetVersionFeatureTypesValue)),
    datasetId: core.serialization.property("dataset_id", core.serialization.string()),
    datasetVersion: core.serialization.property("dataset_version", core.serialization.number()),
    createdOn: core.serialization.property("created_on", core.serialization.number()),
});
