// hume/serialization/resources/customModels/types/ExternalModelVersion.js
"use strict";
/**
 * This file was auto-generated by Fern from our API Definition.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ExternalModelVersion = void 0;
const core = __importStar(require("../../../../core"));
const ExternalModelVersionTag_1 = require("./ExternalModelVersionTag");
const ExternalModelVersionFileType_1 = require("./ExternalModelVersionFileType");
exports.ExternalModelVersion = core.serialization.object({
    id: core.serialization.string(),
    modelId: core.serialization.property("model_id", core.serialization.string()),
    userId: core.serialization.property("user_id", core.serialization.string()),
    version: core.serialization.string(),
    sourceUri: core.serialization.property("source_uri", core.serialization.string()),
    datasetVersionId: core.serialization.property("dataset_version_id", core.serialization.string()),
    createdOn: core.serialization.property("created_on", core.serialization.number()),
    metadata: core.serialization
        .record(core.serialization.string(), core.serialization.record(core.serialization.string(), core.serialization.unknown()))
        .optional(),
    description: core.serialization.string().optional(),
    tags: core.serialization.list(ExternalModelVersionTag_1.ExternalModelVersionTag).optional(),
    fileType: core.serialization.property("file_type", ExternalModelVersionFileType_1.ExternalModelVersionFileType.optional()),
    targetFeature: core.serialization.property("target_feature", core.serialization.string().optional()),
    taskType: core.serialization.property("task_type", core.serialization.string().optional()),
    trainingJobId: core.serialization.property("training_job_id", core.serialization.string().optional()),
});
