// hume/serialization/resources/customModels/types/ModelsPredictions.js
"use strict";
/**
 * This file was auto-generated by Fern from our API Definition.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ModelsPredictions = void 0;
const core = __importStar(require("../../../../core"));
const PredictionsOptionalNullFacePrediction_1 = require("./PredictionsOptionalNullFacePrediction");
const PredictionsOptionalNullBurstPrediction_1 = require("./PredictionsOptionalNullBurstPrediction");
const PredictionsOptionalTranscriptionMetadataProsodyPrediction_1 = require("./PredictionsOptionalTranscriptionMetadataProsodyPrediction");
const PredictionsOptionalTranscriptionMetadataLanguagePrediction_1 = require("./PredictionsOptionalTranscriptionMetadataLanguagePrediction");
const PredictionsOptionalTranscriptionMetadataNerPrediction_1 = require("./PredictionsOptionalTranscriptionMetadataNerPrediction");
const PredictionsOptionalNullFacemeshPrediction_1 = require("./PredictionsOptionalNullFacemeshPrediction");
exports.ModelsPredictions = core.serialization.object({
    face: PredictionsOptionalNullFacePrediction_1.PredictionsOptionalNullFacePrediction.optional(),
    burst: PredictionsOptionalNullBurstPrediction_1.PredictionsOptionalNullBurstPrediction.optional(),
    prosody: PredictionsOptionalTranscriptionMetadataProsodyPrediction_1.PredictionsOptionalTranscriptionMetadataProsodyPrediction.optional(),
    language: PredictionsOptionalTranscriptionMetadataLanguagePrediction_1.PredictionsOptionalTranscriptionMetadataLanguagePrediction.optional(),
    ner: PredictionsOptionalTranscriptionMetadataNerPrediction_1.PredictionsOptionalTranscriptionMetadataNerPrediction.optional(),
    facemesh: PredictionsOptionalNullFacemeshPrediction_1.PredictionsOptionalNullFacemeshPrediction.optional(),
});
