// hume/api/resources/empathicVoice/client/Client.js
"use strict";
/**
 * This file was auto-generated by Fern from our API Definition.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.EmpathicVoice = void 0;
const Client_1 = require("../resources/tools/client/Client");
const Client_2 = require("../resources/prompts/client/Client");
const Client_3 = require("../resources/configs/client/Client");
const Client_4 = require("../resources/chats/client/Client");
const Client_5 = require("../resources/chatGroups/client/Client");
class EmpathicVoice {
    constructor(_options = {}) {
        this._options = _options;
    }
    get tools() {
        var _a;
        return ((_a = this._tools) !== null && _a !== void 0 ? _a : (this._tools = new Client_1.Tools(this._options)));
    }
    get prompts() {
        var _a;
        return ((_a = this._prompts) !== null && _a !== void 0 ? _a : (this._prompts = new Client_2.Prompts(this._options)));
    }
    get configs() {
        var _a;
        return ((_a = this._configs) !== null && _a !== void 0 ? _a : (this._configs = new Client_3.Configs(this._options)));
    }
    get chats() {
        var _a;
        return ((_a = this._chats) !== null && _a !== void 0 ? _a : (this._chats = new Client_4.Chats(this._options)));
    }
    get chatGroups() {
        var _a;
        return ((_a = this._chatGroups) !== null && _a !== void 0 ? _a : (this._chatGroups = new Client_5.ChatGroups(this._options)));
    }
}
exports.EmpathicVoice = EmpathicVoice;
