// hume/dist/serialization/resources/expressionMeasurement/types/Models.js
"use strict";
/**
 * This file was auto-generated by Fern from our API Definition.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Models = void 0;
const core = __importStar(require("../../../../core"));
const Face_1 = require("./Face");
const Unconfigurable_1 = require("./Unconfigurable");
const Prosody_1 = require("./Prosody");
const Language_1 = require("./Language");
const Ner_1 = require("./Ner");
exports.Models = core.serialization.object({
    face: Face_1.Face.optional(),
    burst: Unconfigurable_1.Unconfigurable.optional(),
    prosody: Prosody_1.Prosody.optional(),
    language: Language_1.Language.optional(),
    ner: Ner_1.Ner.optional(),
    facemesh: Unconfigurable_1.Unconfigurable.optional(),
});
