// hume/dist/serialization/resources/expressionMeasurement/types/TrainingBaseRequest.js
"use strict";
/**
 * This file was auto-generated by Fern from our API Definition.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.TrainingBaseRequest = void 0;
const core = __importStar(require("../../../../core"));
const CustomModelRequest_1 = require("./CustomModelRequest");
const Dataset_1 = require("./Dataset");
const Task_1 = require("./Task");
const EvaluationArgs_1 = require("./EvaluationArgs");
const Alternative_1 = require("./Alternative");
exports.TrainingBaseRequest = core.serialization.object({
    customModel: core.serialization.property("custom_model", CustomModelRequest_1.CustomModelRequest),
    dataset: Dataset_1.Dataset,
    targetFeature: core.serialization.property("target_feature", core.serialization.string().optional()),
    task: Task_1.Task.optional(),
    evaluation: EvaluationArgs_1.EvaluationArgs.optional(),
    alternatives: core.serialization.list(Alternative_1.Alternative).optional(),
    callbackUrl: core.serialization.property("callback_url", core.serialization.string().optional()),
    notify: core.serialization.boolean().optional(),
});
