// hume/dist/serialization/resources/empathicVoice/types/ToolErrorMessage.js
"use strict";
/**
 * This file was auto-generated by Fern from our API Definition.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ToolErrorMessage = void 0;
const core = __importStar(require("../../../../core"));
const ErrorLevel_1 = require("./ErrorLevel");
const ToolType_1 = require("./ToolType");
exports.ToolErrorMessage = core.serialization.object({
    code: core.serialization.string().optional(),
    content: core.serialization.string().optional(),
    customSessionId: core.serialization.property("custom_session_id", core.serialization.string().optional()),
    error: core.serialization.string(),
    level: ErrorLevel_1.ErrorLevel.optional(),
    toolCallId: core.serialization.property("tool_call_id", core.serialization.string()),
    toolType: core.serialization.property("tool_type", ToolType_1.ToolType.optional()),
    type: core.serialization.stringLiteral("tool_error"),
});
