// hume/serialization/resources/expressionMeasurement/types/NerPrediction.js
"use strict";
/**
 * This file was auto-generated by Fern from our API Definition.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.NerPrediction = void 0;
const core = __importStar(require("../../../../core"));
const PositionInterval_1 = require("./PositionInterval");
const TimeInterval_1 = require("./TimeInterval");
const EmotionScore_1 = require("./EmotionScore");
exports.NerPrediction = core.serialization.object({
    entity: core.serialization.string(),
    position: PositionInterval_1.PositionInterval,
    entityConfidence: core.serialization.property("entity_confidence", core.serialization.number()),
    support: core.serialization.number(),
    uri: core.serialization.string(),
    linkWord: core.serialization.property("link_word", core.serialization.string()),
    time: TimeInterval_1.TimeInterval.optional(),
    confidence: core.serialization.number().optional(),
    speakerConfidence: core.serialization.property("speaker_confidence", core.serialization.number().optional()),
    emotions: core.serialization.list(EmotionScore_1.EmotionScore),
});
