// hume/dist/serialization/resources/customModels/types/ExternalModel.js
"use strict";
/**
 * This file was auto-generated by Fern from our API Definition.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ExternalModel = void 0;
const core = __importStar(require("../../../../core"));
const ExternalModelVersion_1 = require("./ExternalModelVersion");
exports.ExternalModel = core.serialization.object({
    id: core.serialization.string(),
    name: core.serialization.string(),
    createdOn: core.serialization.property("created_on", core.serialization.number()),
    modifiedOn: core.serialization.property("modified_on", core.serialization.number()),
    totalStars: core.serialization.property("total_stars", core.serialization.number()),
    modelIsStarredByUser: core.serialization.property("model_is_starred_by_user", core.serialization.boolean()),
    archived: core.serialization.boolean(),
    latestVersion: core.serialization.property("latest_version", ExternalModelVersion_1.ExternalModelVersion.optional()),
    isPubliclyShared: core.serialization.property("is_publicly_shared", core.serialization.boolean()),
});
